### PR TITLE
[Refactor] Standardize data entry key names to {type}.{qualifier} format

### PR DESCRIPTION
**Pull request by divyanshsinghvi**
Regularize all inter-stage payload keys across models to use a consistent dot-separated naming convention. Add data_entry_keys.py with legacy key mapping, normalize_keys() compat shim, and validate_key() for enforcement.

PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
Solves a task in #1351  

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan. Please provide the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [ ] The test results. Please paste the results comparison before and after, or the e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user-facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by divyanshsinghvi** [comment]
cc: @tzhouam

**Comment by tzhouam** [comment]
Will take a look

File: vllm_omni/data_entry_keys.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    >>> validate_key("codes.audio")
+    True
+    >>> validate_key("finished")
+    False
+    """
+    parts = key.split(".", 1)
+    return len(parts) == 2 and parts[0] in _VALID_TYPES and len(parts[1]) > 0
+
+
+def normalize_keys(payload: dict[str, Any], *, warn: bool = True) -> dict[str, Any]:
```
Comment (tzhouam) [inline vllm_omni/data_entry_keys.py]: Nice first step on standardizing the key names!

One thought for a follow-up: rather than enforcing naming conventions via string validation at runtime, could we abstract the payload into structured types?

Since the payload needs to be easily serializable across stages, nested `TypedDict`s would be a natural fit — they're plain `dict`s at runtime (zero overhead, trivially serializable), but give us static type checking and IDE autocomplete.

For each category, we can make every known qualifier an explicit field, so misspellings are caught statically:

```python
class HiddenStates(TypedDict, total=False):
    output: Tensor
    trailing_text: Tensor
    last: Tensor
    layers: dict[int, Tensor]  # dynamic layer indices (0, 24, ...)

class Embeddings(TypedDict, total=False):
    prefill: Tensor
    decode: Tensor
    cached_decode: Tensor
    tts_bos: Tensor
    tts_eos: Tensor
    tts_pad: Tensor
    voice: Tensor
    speech_feat: Tensor

class Codes(TypedDict, total=False):
    audio: Tensor
    ref: Tensor

class Ids(TypedDict, total=False):
    all: list[int]
    prompt: list[int]
    output: list[int]
    speech_token: list[int]
    prior_image: list[int]

class OmniPayloadMeta(TypedDict, total=False):
    finished: bool
    left_context_size: int
    override_keys: list[str]
    num_processed_tokens: int
    next_stage_prompt_len: int
    ar_width: int
    eol_token_id: int
    visual_token_start_id: int
    visual_token_end_id: int
    gen_token_mask: Tensor
    generated_len: int
    omni_task: list[str]
    height: int
    width: int

class OmniPayload(TypedDict, total=False):
    hidden_states: HiddenStates
    embeddings: Embeddings
    ids: Ids
    codes: Codes
    meta: OmniPayloadMeta

**Review by tzhouam — COMMENTED** [review]
Looks nice

**Comment by tzhouam** [comment]
@congw729 Can we trigger a Nightly Test for this PR as it modifies many models

File: vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_talker.py [hunk 2]
```diff
@@ -452,18 +453,18 @@ def make_omni_output(self, model_outputs: torch.Tensor | OmniOutput, **kwargs: A
         for info in info_dicts:
             if not isinstance(info, dict):
                 continue
-            ac = info.get("audio_codes")
+            ac = info.get("codes.audio")
```
[the remark this answers is not in the document]
Comment (divyanshsinghvi) [reply vllm_omni/model_executor/models/qwen3_tts/qwen3_tts_talker.py]: fixed.

File: vllm_omni/model_executor/models/mammoth_moda2/mammoth_moda2.py [hunk 3]
```diff
@@ -608,17 +608,18 @@ def _apply_t2i_token_constraints(self, logits: torch.Tensor) -> torch.Tensor:
...
-            ar_width = runtime_info["ar_width"][0]
-            eol_token_id = runtime_info["eol_token_id"][0]
-            visual_start = runtime_info["visual_token_start_id"][0]
-            visual_end = runtime_info["visual_token_end_id"][0]
-            generated_len = runtime_info["generated_len"]
+            ar_width = meta["ar_width"][0]
+            eol_token_id = meta["eol_token_id"][0]
+            visual_start = meta["visual_token_start_id"][0]
+            visual_end = meta["visual_token_end_id"][0]
+            generated_len = meta["generated_len"]
```
[the remark this answers is not in the document]
Comment (divyanshsinghvi) [reply vllm_omni/model_executor/models/mammoth_moda2/mammoth_moda2.py]: fixed.

File: vllm_omni/worker/gpu_model_runner.py [hunk 4]
```diff
@@ -1300,12 +1300,12 @@ def _talker_mtp_forward(self, decode_req_ids: list[str], inputs_embeds: torch.Te
...
         # update the inputs_embeds and code_predictor_codes
         code_predictor_codes_cpu = code_predictor_codes.detach().to("cpu").contiguous()
-        out_key = getattr(self.model, "talker_mtp_output_key", "code_predictor_codes")
+        out_key = getattr(self.model, "talker_mtp_output_key", ("codes", "audio"))
         for idx, req_id in enumerate(decode_req_ids):
             req_index = self.input_batch.req_ids.index(req_id)
             start_offset = int(self.query_start_loc.cpu[req_index])
             inputs_embeds[start_offset : start_offset + 1] = req_embeds[idx : idx + 1]
-            update_dict = {out_key: code_predictor_codes_cpu[idx : idx + 1]}
+            update_dict = {out_key[0]: {out_key[1]: code_predictor_codes_cpu[idx : idx + 1]}}
```
[the remark this answers is not in the document]
Comment (divyanshsinghvi) [reply vllm_omni/worker/gpu_model_runner.py]: Added checks so things should fail before it rather than hidden bug.

**Comment by tzhouam** [comment]
please resolve the conflict brought by the frontend rebase

**Comment by divyanshsinghvi** [comment]
Fixing build errors

**Comment by divyanshsinghvi** [comment]
LGTM

File: vllm_omni/data_entry_keys.py [hunk 5]
```diff
@@ -0,0 +1,235 @@
+"""Structured payload types for inter-stage communication.
```
Comment (hsliuustc0106) [inline vllm_omni/data_entry_keys.py:1]: why we put it here?

**Review by tzhouam — APPROVED** [review]
LGTM
